powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/controller/v3/UpgradeController.java

```diff
@@ -85,7 +85,8 @@ public EciesEncryptedResponse upgradeStart(@RequestBody EciesEncryptedRequest re
         try {
             PowerAuthEncryptionHttpHeaderValidator.validate(header);
         } catch (InvalidPowerAuthHttpHeaderException ex) {
-            throw new PowerAuthUpgradeException(ex.getMessage());
+            logger.warn("Encryption validation failed, error: {}", ex.getMessage());
+            throw new PowerAuthUpgradeException();
```

Comment:
Let's pass `ex` as a parameter to the new exception to keep the cause somewhere.